### PR TITLE
Support jtds driver 2

### PR DESCRIPTION
**Pull request by muga**

### CODE REVIEW

**Comment by frsyuki** [comment]
@muga @uu59 Behavior fix from 61e626f

* instance was wrongly set. jTDS and MS driver use different URL format
* socketTimeout should be set

@hito4t Changes from master:

* Class.forName(...) is used instead of DriverManager so that plugin can tell driver type (and show "Using jTDS Driver" message) before building URL and properties.
* File set at driver_path must exist. Otherwise validation fails.
* Setting `integratedSecurity: false` is allowed when `url` is set (it became error before). This is useful to when people generate config programmably with automatically embedding default values.
* `'port: {}' option is ignored because instance option is set` message is shown when `instance` is set

Setting driver_path means setting MS's driver path. it's impossible to override jTDS driver path (because it's included in plugin package).

**Comment by uu59** [comment]
Thanks following up my pull request.

**Comment by muga** [comment]
thank you. @hito4t @frsyuki when you can, please merge it into master branch. It would be great.

File: embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java [hunk 1]
```diff
@@ -1,21 +1,26 @@
 package org.embulk.input;
 
 import java.sql.Connection;
+import java.sql.Driver;
 import java.sql.DriverManager;
```
Comment (hito4t) [inline embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java]: This import is unused.

File: embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java [hunk 2]
```diff
@@ -66,81 +71,186 @@
...
+        if (sqlServerTask.getUrl().isPresent()) {
+            if (sqlServerTask.getHost().isPresent()
+                    || sqlServerTask.getInstance().isPresent()
+                    || sqlServerTask.getDatabase().isPresent()
+                    || sqlServerTask.getIntegratedSecurity()) {
+                throw new ConfigException("'host', 'port', 'instance', 'database' and 'integratedSecurity' options are invalid if 'url' option is set.");
+            }
+
+            if (sqlServerTask.getUser().isPresent()) {
+                props.setProperty("user", sqlServerTask.getUser().get());
```
Comment (hito4t) [inline embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java]: 'user' is already set in line 148.

File: embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java [hunk 3]
```diff
@@ -66,81 +71,186 @@
...
+            // integratedSecutiry is not supported, user + password is required
+            if (sqlServerTask.getIntegratedSecurity()) {
+                throw new ConfigException("'integratedSecutiry' option is not supported with jTDS driver. Set 'driver_path: /path/to/sqljdbc.jar' option if you want to use Microsoft SQLServerDriver.");
+            }
+
+            if (!sqlServerTask.getUser().isPresent()) {
+                throw new ConfigException("'user' option is required but not set.");
+            }
+
+            props.setProperty("user", sqlServerTask.getUser().get());
```
Comment (hito4t) [inline embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java]: 'user' is already set in line 148.

File: embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java [hunk 4]
```diff
@@ -66,81 +71,186 @@
...
+                if (sqlServerTask.getUser().isPresent()) {
+                    throw new ConfigException("'user' options are invalid if 'integratedSecutiry' option is set.");
+                }
+                props.setProperty("integratedSecurity", "true");
+            }
+            else {
+                if (!sqlServerTask.getUser().isPresent()) {
+                    throw new ConfigException("'user' option is required but not set.");
+                }
+                props.setProperty("user", sqlServerTask.getUser().get());
```
Comment (hito4t) [inline embulk-input-sqlserver/src/main/java/org/embulk/input/SQLServerInputPlugin.java]: 'user' is already set in line 148.

**Comment by hito4t** [comment]
Thank you for the PR!
This feature is useful because users doesn't need to download JDBC driver and set driver_path.
I've added some comments, please check them.

**Comment by uu59** [comment]
@hito4t Thanks reviewing.
I've pushed commits based on your comments.

**Comment by uu59** [comment]
@hito4t Please merge this if you feel okay when you have a time.

**Comment by muga** [comment]
👍

**Comment by uu59** [comment]
Thanks merging. Would you have a plan to release 0.7.3?

**Comment by hito4t** [comment]
I'll release 0.7.3 in this week.

**Comment by uu59** [comment]
Good to hear! thanks 🙏

**Comment by hito4t** [comment]
I've released!

**Comment by uu59** [comment]
Thank you ❤️

**Pull request merged**